src/goto-instrument/contracts/dynamic-frames/dfcc_loop_tags.cpp

```diff
@@ -0,0 +1,132 @@
+/*******************************************************************\
+
+Module: Dynamic frame condition checking
+
+Author: Qinheping Hu, qinhh@amazon.com
+
+Author: Remi Delmas, delmasrd@amazon.com
+
+Date: March 2023
+
+\*******************************************************************/
+
+#include "dfcc_loop_tags.h"
+
+#define ID_LOOP_ID "loop-id"
+#define ID_LOOP_HEAD "loop-head"
+#define ID_LOOP_BODY "loop-body"
+#define ID_LOOP_LATCH "loop-latch"
+#define ID_LOOP_EXITING "loop-exiting"
+#define ID_LOOP_TOP_LEVEL "loop-top-level"
+
+void dfcc_set_loop_id(
+  goto_programt::instructiont &instruction,
+  const std::size_t loop_id)
+{
+  instruction.source_location_nonconst().set(ID_LOOP_ID, loop_id);
```

Comment:
Is the id here the loop number or the id computed in the nesting graph? I suppose it is the later one. We may want to rename it to `ID_LOOP_NESTING_ID` or leave a comment to highlight the difference between the `goto-program/loop_ids.h` and the id defined here.

Reply:
In this module the loop_id has no built-in meaning, it only gets one in the context where tagging gets used, so I'd say this is orthogonal to the loop_nesting_graph.